docs/Design/open telemetry.md

```diff
@@ -0,0 +1,218 @@
+  },
+  status: { code: 0 },
+  events: [],
+  links: []
+}
+{
+  traceId: 'c165a7a72532438262039f0ca57e57c4',
+  parentId: undefined,
+  traceState: undefined,
+  name: 'POST',
+  id: '2cb405fe86791bf0',
+  kind: 1,
+  timestamp: 1683667014784000,
+  duration: 707563,
+  attributes: {
+    'http.url': 'http://localhost:3000/local/v3.3b/ed-fi/students/',
+    'http.host': 'localhost:3000',
+    'net.host.name': 'localhost',
+    'http.method': 'POST',
+    'http.scheme': 'http',
+    'http.target': '/local/v3.3b/ed-fi/students/',
+    'http.user_agent': 'vscode-restclient',
+    'http.request_content_length_uncompressed': 204,
+    'http.flavor': '1.1',
+    'net.transport': 'ip_tcp',
+    'net.host.ip': '127.0.0.1',
+    'net.host.port': 3000,
+    'net.peer.ip': '127.0.0.1',
+    'net.peer.port': 57641,
+    'http.status_code': 201,
+    'http.status_text': 'CREATED'
+  },
+  status: { code: 0 },
+  events: [],
+  links: []
+}
+
+```
+
+The first trace coming from the POST to the students endpoint to create a
+student. The second trace is the call adding the new student to Opensearch
+
+The above trace shows us what Node knows about what happened with Meadowlark.
+Which is that a POST request to the students endpoint was made, a resource was
+created, and a PUT call made to OpenSearch, which also resulted in a 201.
+
+It is possible to add attributes to traces by retrieving the current trace and
+adding new attributes through the OTel SDK.
+
+OTel has an extensive [collection of
+plugins](https://opentelemetry.io/ecosystem/registry/?language=js&component=instrumentation)
+to auto instrument JavaScript packages, including a number of packages that are
+used in Meadowlark including Fastify, MongoDB and PostgeSQL.
+
+## OTel Collector
+
+There are a couple of ways to handle trace data after it's been created. One way
+is to directly send telemetry data to a backend such as Prometheus, Jager or
+ElasticSearch from Meadowlark.
+
+OTel also provides the Collector. The Collector offers a vendor-agnostic
+implementation of how to receive, process and export telemetry data.
+
+Pros of using a collector:
+
+- Easy to set up (docker container available)
+- Can be run as a cluster
+- Duplicate telemetry data from the collector to multiple endpoints
+- Collects from multiple services before moving to telemetry backend
+  - Process the data (adding/removing attributes, handling batching, etc.)
+    before sending it to another destination
+- Decouple the producer with the consumer
+
+Due to the flexibility of how the collector can be set up and the control it
+gives to it's users, collectors are the suggested way to handle OTel output.
+
+# How Meadowlark could use Open Telemetry
+
+Since we already have a form of tracing within Meadowlark through our existing
+logs, it doesn't seem like it would be all that difficult to add OTel tracing to
+Meadowlark, although managing parent/child traces may be a bit of work to do
+properly.
```

Comment:
What tangible benefit would we get from switching from our home-grown tracing to Open Telemetry?

Reply:
Will add thoughts on this to reasearch